proof-systems/src/darlin/tests/simple_marlin.rs

```diff
@@ -0,0 +1,161 @@
+//! A R1CS density one test circuit of specified number of constraints, which processes 
+//! two public inputs satisfying a simple quadratic relation.
+use algebra::{Field, AffineCurve, UniformRand};
+use r1cs_core::{ConstraintSynthesizer, ConstraintSystem, SynthesisError};
+use poly_commit::ipa_pc::{InnerProductArgPC, CommitterKey, UniversalParams};
+use marlin::{
+    Marlin, ProverKey as MarlinProverKey, VerifierKey as MarlinVerifierKey,
+};
+use crate::darlin::pcd::{
+    PCDParameters, simple_marlin::SimpleMarlinPCD
+};
+use rand::{ Rng, RngCore };
+use digest::Digest;
+use std::ops::MulAssign;
+
+/// A simple test circuit with two field elements c,d as inputs, enforced to satisfy
+///     (c,d) = a*(b,b^2),
+/// To produce the given `num_constraints`, the same two constraints
+///     a * b = c
+///     c * b = d
+/// are repeated accordingly. To acchieve the give `num_variables`, a corresponding number of
+/// dummy witness variables are allocated.
```

Comment:
This circuit could be dangerous for benching (If we ever intend to do that, I have to check it once again).

Reply:
Nope, we just use it for tests

Reply:
I spent another thought on it. Even if we just use it for unit tests, it is not very representative. Its R1CS matrices are very low rank. (However, it helped us to find the mistake of wront "zipping" for an iterator). We could keep it, but call it low_rank_test, and add a representative by copy-past TestCircuit1c for the Tweedles?